### PR TITLE
 RID work for macOS 12

### PR DESCRIPTION
**Pull request by wfurt**
Note this is weird as with 11.0 the naming has changed as well as and OS was lying us of a while about actual os version, reporting 10.17 instead. Since we seems to use updated SDK this does not seems to be necessary any more so as the game of mapping kernel to OS version.

It seems like starting with 11.0, the releases we used to track as 10.x are really now major releases and the 'x' becomes insignificant. So I changed the 12 rid to match Win & RHEL.

We may do more cleanup for Environment.OSVersion but since that seems to war I'll do it separately if needed. 
`kern.osproductversion` was missing in some very old OS X versions but it is available in 10.13 and later.

### CODE REVIEW

File: src/native/corehost/hostmisc/pal.unix.cpp [hunk 1]
```diff
@@ -549,48 +549,30 @@ pal::string_t pal::get_current_os_rid_platform()
...
-                ridOS.append(pal::to_string(majorVersion - 20));
+                // strip anything after second dot and return something like 10.5
+                *pos = 0;
+                size = pos - str;
             }
         }
+
+        std::string release(str, size);
+        ridOS.append(_X("osx."));
+        ridOS.append(release);
```
Comment (am11) [inline src/native/corehost/hostmisc/pal.unix.cpp:575]: a nit: avoid allocation and align with existing style, e.g. in FreeBSD case below:

```suggestion
        ridOS.append(_X("osx."));
            .append(str, size);
```

**Comment by wfurt** [comment]
ping. anybody brave?

**Comment by marek-safar** [comment]
Are you planning to backport this to 6.0?

**Comment by wfurt** [comment]
yes. I think this would qualify for servicing. I will get ready separate change for 5.x

**Comment by marek-safar** [comment]
@agocke please speed up the review

**Comment by agocke** [comment]
The precedent we're setting with Windows 11 is that we match the OS product version, not the marketing version. It looks to me like that's what this PR is doing -- that the OS is reporting a new product version. Is that correct?

**Comment by wfurt** [comment]
There is really no marketing version here. With this, we should match what `sw_vers` reports and what you see one the `About` dialog @agocke.

There are two parts: In the old days, there was no convent API to get that so we had mapping from kernel version to OS version. AFAIK that is no longer needed. 
For bits but agains older SDK the OS would lie and report "compact" version eg. 10.x instead of 11.0 and 12.0. That impacts 5.0 (will need separate change) but not 6.0 as far as I can tell. 

And there is the "normal" rid work to add 12.0 to the graph.

**Comment by vitek-karas** [comment]
I personally don't know versioning story around macOS.

On Windows we decided (See https://github.com/dotnet/runtime/pull/58588) that there's little point in introducing win11 RIDs. Core of the reasoning is that OS still returns 10 as the major version and that we don't introduce new RIDs for Windows Server releases either.

If on macOS the OS actually returns the new version and it makes a material difference, then we should add a new RID (I don't know this though). I guess this is just a "small pushback" to make sure that we need to add the new RID (trying to keep the always growing list of RIDs in check).

**Comment by wfurt** [comment]
We already made the leap with 11.0 @vitek-karas. (with 5.0 & ARM based machines)

**Comment by agocke** [comment]
It looks like Big Sur reported itself as 11.0, except if you set `SYSTEM_VERSION_COMPAT=1`. I don't think we should operate based on compat versions, so by that metric, the system version was 11.0 and Monterey should be 12.0.

Change LGTM

**Review by agocke — APPROVED** [review]

**Comment by vitek-karas** [comment]
OK - makes sense :smile:

**Pull request merged**